### ISSUE TITLE
Non breaking spaces before citations

### ISSUE THREAD

**Issue by domoritz** [uwdata/living-papers#106]
It would be great if living papers could render non breaking spaces before citations to get a smoother reading flow and avoid this:

<img width="694" alt="Screenshot 2024-10-15 at 20 29 36" src="https://github.com/user-attachments/assets/7876c3c6-2f1a-4a3b-aa6b-ff3aece02af6">

In latex, it would compile to `~`.

**Comment by jheer** [comment]
Yes that would be nice. The design question here is what should the syntax be? For HTML it is `&nbsp;` which is not the nicest but would work. We don’t want to use `~` as we want to preserve straightforward use of tilde as a character. Any other ideas?

**Comment by jheer** [comment]
We could try to do this automatically before citations but I’m not sure that’s something one always wants to do, especially across varied citation styles.

**Comment by domoritz** [comment]
What are cases where you don't want it? Long citation styles?

As a simple step, you could support the ` ` (non breaking space char, option + space) and convert it to ~ in latex (or maybe panda does that already). The bad thing is that it's pretty much indistinguishable from a normal space in the markdown. I can't think of a good symbol for non breaking spaces. Typst also uses ~. 

Another odd thing I found were breaking spaces in `Figure X` references but that's probably an easier fix.

<img width="666" alt="Screenshot 2024-10-15 at 23 32 25" src="https://github.com/user-attachments/assets/73f0d3f0-ffd8-4db3-b176-04a3f5c1f6c5">
